agent/config/config.go

```diff
@@ -571,9 +573,10 @@ type AutoEncrypt struct {
 type Connect struct {
 	// Enabled opts the agent into connect. It should be set on all clients and
 	// servers in a cluster for correct connect operation.
-	Enabled    *bool                  `json:"enabled,omitempty" hcl:"enabled" mapstructure:"enabled"`
-	CAProvider *string                `json:"ca_provider,omitempty" hcl:"ca_provider" mapstructure:"ca_provider"`
-	CAConfig   map[string]interface{} `json:"ca_config,omitempty" hcl:"ca_config" mapstructure:"ca_config"`
+	Enabled                         *bool                  `json:"enabled,omitempty" hcl:"enabled" mapstructure:"enabled"`
+	CAProvider                      *string                `json:"ca_provider,omitempty" hcl:"ca_provider" mapstructure:"ca_provider"`
+	CAConfig                        map[string]interface{} `json:"ca_config,omitempty" hcl:"ca_config" mapstructure:"ca_config"`
+	MeshGatewayWANFederationEnabled *bool                  `json:"enable_mesh_gateway_wan_federation" hcl:"enable_mesh_gateway_wan_federation" mapstructure:"enable_mesh_gateway_wan_federation"`
```

Comment:
Q: is this name too verbose?

Reply:
Maybe just - `WANFedEnabled` or `WANFederationEnabled` - do we need the Mesh Gateway part?

Reply:
Well technically normal multidc consul _is_ federation via the wan.

Reply:
It is a bit verbose but I can't think of a shorter version that actually communicates the effect of setting it correctly. We could shorten `Federation` to `Fed` but in general in HashiCorp products our convention is not to abbreviate name or config options. So I guess leave it like this!

Reply:
Do we need WAN? What other type of federation is there?